File: root/rest/src/main/java/org/sc/common/rest/PoiDto.java
package org.sc.common.rest;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.sc.data.model.PoiMacroType;

import java.util.List;

@Data
@AllArgsConstructor
public class PoiDto {
    private String id;
    private String name;
    private String description;
    private List<String> tags;
    private PoiMacroType macroType;
    private List<String> microType;
    private List<LinkedMediaDto> mediaList;
    private List<String> trailIds;
    private CoordinatesDto coordinates;
    private List<String> externalResources;
    private List<KeyValueDto> keyVal;
    private RecordDetailsDto recordDetails;
    private String externalId;
    private String externalSystemName;
}

File: root/common/src/main/java/org/sc/common/rest/PoiDto.kt
package org.sc.common.rest

import java.util.*

data class PoiDto (val id: String,
                   val name: String,
                   val description: String,
                   val tags: List<String>,
                   val macroType: PoiMacroType,
                   val microType: List<String>,
                   val mediaIds: List<String>,
                   val trailIds: List<String>,
                   val trailCoordinates: TrailCoordinates,
                   val createdOn: Date,
                   val lastUpdatedOn: Date)